### PR TITLE
Initial development of the pit form

### PR DESCRIPTION
**Pull request by DanielLaszczych**
Added the last tab you were into local storage
Added a reload alert in case someone accidentally refreshes a page 
Started working on the pit form

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -9,6 +9,7 @@
     "@testing-library/user-event": "^7.2.1",
     "bootstrap": "^4.4.1",
     "cors": "^2.8.5",
+    "express": "^4.17.1",
```
Comment (sharonkass) [inline client/package.json:12]: Is this necessary? The parent package.json already has express. And I see you bumped the express version in the parent package.json file but here in this client package.json it is the older version. This is the risk of having this in both places.

File: client/src/App.js [hunk 2]
```diff
@@ -1,55 +1,65 @@
...
 
 class App extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      apiResponse:"",
-      selectedTab: ''
-    };
-  }
+  state = {
```
Comment (sharonkass) [inline client/src/App.js:26]: Why was this taken out of a constructor?

File: client/src/components/PitContent.js [hunk 3]
```diff
@@ -1,12 +1,109 @@
-import React, {Component} from 'react'
+import React, { Component } from "react";
+import Form from "react-bootstrap/Form";
+import Button from "react-bootstrap/Button";
+import Row from "react-bootstrap/Row";
+import Col from "react-bootstrap/Col";
+import "./PitContent.css";
 
 class PitContent extends Component {
+  state = {
```
Comment (sharonkass) [inline client/src/components/PitContent.js:11]: Same comment here. I think a constructor might be a good practice. Let's check with Jon.

File: client/src/components/PitContent.js [hunk 4]
```diff
@@ -1,12 +1,109 @@
...
+      event.preventDefault();
+      event.stopPropagation();
+    }
+
+    this.setState({ validated: true });
+  };
+
+  render() {
+    return (
+      <Form
```
Comment (sharonkass) [inline client/src/components/PitContent.js]: Nice job on the pit data collection!

File: client/src/components/ScoutContent.js [hunk 5]
```diff
@@ -1,64 +1,56 @@
...
-import Row from 'react-bootstrap/Row';
-import Col from 'react-bootstrap/Col';
+import React, { Component } from "react";
+import Form from "react-bootstrap/Form";
+import Button from "react-bootstrap/Button";
+import Row from "react-bootstrap/Row";
+import Col from "react-bootstrap/Col";
 
 class ScoutContent extends Component {
-
```
Comment (sharonkass) [inline client/src/components/ScoutContent.js:8]: I'm not sure why these are showing as diffs. Maybe just some newline issues. Not a big deal, makes it hard to see diffs, but let's not worry about it.

File: client/src/components/TabNav.js [hunk 6]
```diff
@@ -1,31 +1,35 @@
...
+import Tab from "react-bootstrap/Tab";
 
 class TabNav extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      activeTab: props.activeTab || "pit"
-    };
-   this.handleSelect = this.handleSelect.bind(this)
+  state = {
```
Comment (sharonkass) [inline client/src/components/TabNav.js:6]: Same question about constructor.

File: package-lock.json [hunk 7]
```diff
@@ -17,6 +17,11 @@
         "@types/babel-types": "*"
       }
     },
+    "abbrev": {
```
Comment (sharonkass) [inline package-lock.json:20]: There are lots of changes to this file. What are all these for?

**Review by sharonkass — COMMENTED** [review]
Nice job on this pull request! I have a few questions though - these are for my own education and are not criticisms but depending on answers/discussion some small changes may be needed.

File: client/package.json (hunk 1 above)
Comment (JonathanWexler) [reply client/package.json:12]: I agree here. @DanielLaszczych, is there a reason to add `express` here? I think we should be fine removing express from the client's `package.json` and only using it in the server's `package.json`

File: client/src/App.js (hunk 2 above)
Comment (JonathanWexler) [reply client/src/App.js:26]: Let's test this to make sure the state is initialized on `this` before we set new values to `apiResponse` and `selectedTab`. This should still work but might affect React's _reactivity_

File: client/src/components/PitContent.js (hunk 3 above)
Comment (JonathanWexler) [reply client/src/components/PitContent.js:11]: same as above. Removing constructor is ok as default super is called, but let's make sure of the convention for setting initial state properties.

File: client/src/components/ScoutContent.js (hunk 5 above)
Comment (JonathanWexler) [reply client/src/components/ScoutContent.js:8]: looks like spacing was adjusted. Seems fine. We should create a linter file for whether to use double or single quotations. This way your compiler(webpack)/linter will complain when using them the wrong way.

File: client/src/components/TabNav.js (hunk 6 above)
Comment (JonathanWexler) [reply client/src/components/TabNav.js:6]: same thoughts as mine above

File: package-lock.json (hunk 7 above)
Comment (JonathanWexler) [reply package-lock.json:20]: this is auto generated, should be fine with the package.json changes

**Review by JonathanWexler — APPROVED** [review]

**Pull request merged**